src/JustSaying/Messaging/Monitoring/IMessageMonitor.cs

```diff
@@ -3,16 +3,69 @@
 
 namespace JustSaying.Messaging.Monitoring
 {
+    /// <summary>
+    /// Provides a mechanism for monitoring when various message actions are
+    /// performed in JustSaying. This is typically used for metrics gathering.
+    /// </summary>
     public interface IMessageMonitor
     {
+        /// <summary>
+        /// Monitors when a specific message type threw an exception while being handled
+        /// by a subscriber.
+        /// </summary>
+        /// <param name="messageType">The type of the message that was being handled.</param>
         void HandleException(Type messageType);
+
+        /// <summary>
+        /// Monitors when a particular message that threw an exception while being handled
+        /// by a subscriber.
+        /// </summary>
+        /// <param name="ex">The exception that was thrown.</param>
+        /// <param name="message">The message being handled when the exception was thrown.</param>
         void HandleError(Exception ex, Message message);
```

Comment:
Similar again language-wise, though this method is called if something goes wrong in the dispatcher when deserialising the `TMessage` , so we don't have a TMessage to call `HandleException` with.

Perhaps:

`Called when an error occurs while attempting to deserialise a message from the `Amazon.SQS.Model.Message.` ?

The `message` parameter in this case is actually the raw SQS message and not a JustSaying model.